src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -912,13 +921,40 @@ export default class DataTable extends Vue {
     this.slicedData = this.sliceData(this._sortedData || this._serializedDataBody);
   }
 
+  _showSelectedRows(isSelected: boolean) {
+    const selectedRow = this._serializedDataBody.filter((row: DataTableRow) => {
+      return this.selectedRows.includes(row.rowId);
+    });
+
+    for (const i in selectedRow) {
+      selectedRow[i].selected = true;
+    }
+
+    if (isSelected) {
+      this.slicedData = selectedRow;
+    } else {
+      const selectedRowsData = this.sliceData(this._sortedData || this._serializedDataBody);
```

Comment:
This would be just the first page of the whole data set (selected + not selected) provided to the data table.
```suggestion
      this.slicedData = this.sliceData(this._sortedData || this._serializedDataBody);
```